### PR TITLE
fix(): sleep before redirect

### PR DESCRIPTION
**Pull request by AMoreaux**
## Summary
This Pull Request centralizes the redirection logic by introducing a reusable `useRedirect` hook, which replaces direct usage of `window.location.href` with more standardized and testable functionality across multiple modules.

- Introduced a new `useRedirect` hook for handling redirection logic with optional controlled delays.
- Refactored redirection implementations in various modules (`useAuth`, workspace, and settings-related hooks, etc.) to use the newly introduced `useRedirect` or related high-level hooks.
- Updated API and documentation to include or improve support for SSO, particularly OIDC and SAML setup processes in server logic.
- Enhanced frontend and backend configurability with new environment variable settings for SSO.

### CODE REVIEW

File: packages/twenty-front/src/modules/domain-manager/hooks/useRedirect.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Don't use this hook directly! Prefer the high level hooks like:
+// useRedirectToDefaultDomain and useRedirectToWorkspaceDomain
+import { sleep } from '~/utils/sleep';
+
+export const useRedirect = () => {
+  const redirect = (url: string) => {
+    // This hacky workaround is necessary to ensure the tokens stored in the cookie are updated correctly.
+    sleep(0).then(() => {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/domain-manager/hooks/useRedirect.ts:8]: useDebounced (100ms)

Comment (charlesBochet) [reply packages/twenty-front/src/modules/domain-manager/hooks/useRedirect.ts:8]: const onMouseLeaveBoard = useDebouncedCallback(() => {
    if (isCompactModeActive && isCardExpanded) {
      setIsCardExpanded(false);
    }
  }, 800);

Comment (charlesBochet) [reply packages/twenty-front/src/modules/domain-manager/hooks/useRedirect.ts:8]: name the hook differently as it's not a generic hook (it's tied to domain redirection): useDomainRedirect

**Review by FelixMalfait — APPROVED** [review]
Great!

**Pull request merged**